### PR TITLE
fix: update angular & angular material to `next` version

### PR DESCRIPTION
**Pull request by mmalerba**

### CODE REVIEW

**Review by wagnermaciel — APPROVED** [review]
LGTM with a couple of nits:

The sidenav lists all have their text uncentered
https://material2-docs-dev.web.app/components/categories

The mat-mdc-cards for the guides look pretty messed up
https://material2-docs-dev.web.app/guides

Not sure if we want to fix these now or later

File: .browserslistrc [hunk 1]
```diff
@@ -1,17 +0,0 @@
-# This file is used by the build system to adjust CSS and JS output to support the specified browsers below.
```
Comment (wagnermaciel) [inline .browserslistrc:1]: Also, why delete this file? Is it no longer used?

Comment (mmalerba) [reply .browserslistrc:1]: I don't know, I just ran `ng update` and it deleted it

**Comment by mmalerba** [comment]
https://material2-docs-dev.web.app doesn't reflect this PR. This PR is just `ng update`-ing to v15. the dev site has this PR + another change to `ng generate` to MDC-based components, and yes that part needs some CSS tweaks

**Pull request merged**